soc/arc/snps_arc_axs/soc.c

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright (c) 2019 Synopsys, Inc. All rights reserved.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * This module provides routines to initialize and support soc-level hardware
+ * for the AXS103 Board.
+ */
+#include <device.h>
+#include <init.h>
+#include "soc.h"
+
+#define	ICTL_INTNO_NUM	28
+#define	CPU_ICTL_INTNO_NUM	5
+#define	NUM_INT_ALL	(CPU_ICTL_INTNO_NUM + ICTL_INTNO_NUM)
+
+#define	REGBASE_ICTL_CPU_INSTATUS	(0xF0002040U)
+
+typedef void (*INT_HANDLER) (void *ptr);
+
+static void axs_103_int_handler_default(void *ptr)
+{
+	ptr = ptr;
+}
+
+static INT_HANDLER axs_103_int_handler_table[NUM_INT_ALL] = {
+	[0 ... NUM_INT_ALL - 1] = axs_103_int_handler_default
+};
+
+static void ictl_gpio_int_isr(void *arg)
+{
+	u32_t status;
+	u32_t i;
+
+	status = *((volatile u32_t *)REGBASE_ICTL_CPU_INSTATUS);
+
+	for (i = 0; i < ICTL_INTNO_NUM; i++) {
+		if (status & (1 << i)) {
+			axs_103_int_handler_table[i + ICTL_INTNO_NUM]((void *)i);
+		}
+	}
+}
+static int arc_axs_init(struct device *dev)
+{
+	ARG_UNUSED(dev);
+
+	IRQ_CONNECT(24, 1, ictl_gpio_int_isr, NULL, 0);
+	irq_enable(24);
+	return 0;
+}
+
+/*SYS_INIT(arc_axs_init, PRE_KERNEL_1, CONFIG_KERNEL_INIT_PRIORITY_DEFAULT);*/
```

Comment:
Shippable fails on this: 'arc_axs_init' defined but not used [-Werror=unused-function].
If interrupt support is still work in progress, better not include it at all.

Reply:
removed